scipy/ndimage/tests/test_filters.py

```diff
@@ -911,6 +920,38 @@ def test_filter_axes(self, filter_func, extra_args, size0, size, axes, xp):
         expected = filter_func(array, *extra_args, all_sizes)
         xp_assert_close(output, expected)
 
+    @pytest.mark.parametrize('func', [ndimage.correlate, ndimage.convolve])
+    @pytest.mark.parametrize('dtype', [np.float32, np.complex64])
+    @pytest.mark.parametrize(
+        'axes', tuple(itertools.combinations(range(-3, 3), 2))
+    )
+    @pytest.mark.parametrize('origin', [(0, 0), (-1, 1)])
+    def test_correlate_convolve_axes(self, func, dtype, axes, origin, xp):
+        if is_cupy(xp):
+            pytest.xfail("https://github.com/cupy/cupy/pull/8339")
+
+        array = np.arange(6 * 8 * 12, dtype=dtype).reshape(6, 8, 12)
+        weights = np.arange(3 * 5).reshape(3, 5)
```

Comment:
```suggestion
        weights_np = np.arange(3 * 5).reshape(3, 5)
```

Reply:
I changed both array and weights to use `xp.arange` during the rebase.